compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -2200,10 +2200,12 @@ impl<D: Decoder> Decodable<D> for EncodedMetadata {
         let len = d.read_usize();
         let mmap = if len > 0 {
             let mut mmap = MmapMut::map_anon(len).unwrap();
+            let mut num_bytes = 0;
             for _ in 0..len {
-                (&mut mmap[..]).write(&[d.read_u8()]).unwrap();
+                num_bytes += (&mut mmap[..]).write(&[d.read_u8()]).unwrap();
             }
             mmap.flush().unwrap();
+            debug_assert!(len == num_bytes);
```

Comment:
Does this seem useful? Should I keep it or revert it? :thinking:

Reply:
Can we use write_all instead (or whatever that was called that ensures the entire slice is written)